WIP Gemini 3h guide updates (#401)
subspace/subspace-docs: docs/farming-&-staking/staking/operators/tips-operator.mdx

import Link from '@docusaurus/Link';
import styles from '@site/src/pages/index.module.css';

### Create operator key (alternative, less secure way):

> An operator needs a key pair to participate in bundle production.

Reviewer: This is still true?

Author: No this is no longer true, ut's been deleted.

Reviewer: I mean the deleted statement "An operator needs a key pair to participate in bundle production." is still true but we've removed it?

Reviewer 2: :100: needs a key pair

Author: Sure, the operator of course needs a key pair. It's just the phrase itself was a preamble of a keypair generation, which is no longer relevant because the way the keys are generated is completely new. Now there's no preferred, or backup method of generating the keypair, so the whole section in the "tips-operator" is dropped, alongside with the sentence.

Author: I moved the phrase to `register-operator.md` under the **Create operator key** section.

> This might be useful if you decicded to switch domains and already have **the secret phrase**. Read more about switching domain in the next section.

Reviewer: ```suggestion
This might be useful if you decided to switch domains and already have **the secret phrase**. Read more about switching domain in the next section.
```

Reviewer 2: I'd say "or already have" because keys can be generated externally.

Reviewer: > I'd say "or already have" because keys can be generated externally.

Updated.

To insert the key, use the following command:

```bash
target/production/subspace-node domain key insert \

> --base-path PATH_TO_NODE --domain-id DOMAIN_ID --keystore-suri "<Secret phrase>" --key-type oper --scheme sr25519

Reviewer 2: This is incorrect, there is no `--key-type` or `--scheme` options. We only support very specific keys and our custom commands uses them implicitly without need/ability to specify.

You can create a key using the following command:

```bash

> target/production/subspace-node run key generate --scheme sr25519

Reviewer 3: @nazar-pc should this work? When I try to `./subspace-node run key ...` I get an `error: unexpected argument 'key' found`.

Reviewer 2: No, `run x` doesn't make sense, that command is for running exclusively. All key-related commands are gone except importing, but now I see that we need to bring generation back. I'll start a Slack thread about this to figure out what the interface should look like,

:::

:::note

> You need to move your keypair to a new location, if you switch domains. Follow the [insert key to keystore section](#insert-key-to-keystore)

Reviewer: ```suggestion
You need to move your keypair to a new location, if you switch domains. Follow the [insert key to keystore section](#insert-key-to-keystore).
```

> You need to update the move your keypain to a new location, if you switch domains. Follow the [insert key to keystore section](#insert-key-to-keystore)

Reviewer: ```suggestion
You need to move your keypair to a new location if you switch domains. Follow the [insert key to keystore section](#insert-key-to-keystore) section.
```

Author: keypain, probably how I was feeling at the moment 😆

Author: Didn't include the word "section" as it's already included in the link.

      - "0.0.0.0:40333:40333/tcp"
    restart: unless-stopped
    command: [
      "run",
      "--chain", "gemini-3g",
      "--base-path", "/var/subspace",
      "--blocks-pruning", "256",
      "--state-pruning", "archive-canonical",
      "--listen-on", "30333",
      "--dsn-listen-on", "/ip4/0.0.0.0/udp/30433/quic-v1",
      "--dsn-listen-on", "/ip4/0.0.0.0/tcp/30433",
      "--rpc-cors", "all",
      "--rpc-methods", "unsafe",

>       "--rpc-listen-on",

Reviewer 2: Here is and in many other places syntax is wrong, this is not a boolean option

Author: Sorry, what we should replace `--rpc-external` with in this case? I thought it could be used with or without specifying the port, as it has a default value unless specified explicitly.

Reviewer 2: Definitely not. `--rpc-listen-on` replaces a bunch of options, it is not a simple renaming. `--help` explains the syntax of each option.

`--rpc-listen-on` is not a port, it is `IP:port`. If you want to listen on all IPs then you specify `0.0.0.0` as IP explicitly. As to port - you specify the port you want, probably `9944`, so the value is `0.0.0.0:9944` here. In future we'll make RPC optional (Substrate currently always starts it unconditionally) and will not run at all unless this option is specified.
